### PR TITLE
Implemented native Router middleware from express

### PR DESCRIPTION
**Pull request by Dalphus**
Biggest change is the addition of `routes.js` to handle all api requests
every other change was made to adapt the codebase to the new framework, hence why there are so many lines changed

also related products no longer renders, but don't worry about that

### CODE REVIEW

**Comment by Dalphus** [comment]
I promise this is my last PR over 100 lines

**Comment by Dalphus** [comment]
probably

**Review by MarZmu — APPROVED** [review]

**Review by MarZmu — APPROVED** [review]

File: client/src/components/Setter.jsx [hunk 1]
```diff
@@ -20,84 +15,36 @@ class Setter extends React.Component {
...
       setReviews,
     } = this.props;
 
-    axios.get('/products/40348')
+    axios.get('/a/products/40348')
       .then((response) => {
-      //console.log(response.data);
         setProductInfo(response.data);
       })
       .catch(() => console.log(`Error loading product info`));
```
[the remark this answers is not in the document]
Comment (Dalphus) [reply client/src/components/Setter.jsx:26]: I think this is Andrew's axios request, I'll bug him for permission to change it

File: server/index.js [hunk 2]
```diff
@@ -1,71 +1,12 @@
...
-});
-
-
-//ALL POST REQUESTS
-app.post('/', (req, res) => {
-  // 2nd arg
-  controllers.post(req.path, req.body.data)
-    .then((questions) => res.status(200).send(questions));
-});
+app.use('/a', routes);
```
[the remark this answers is not in the document]
Comment (Dalphus) [reply server/index.js:9]: the purpose of adding `/a` to the beginning of API requests here is to separate them from other endpoints our server might have in the future.
When I implement `app.all` I'll use something like `app.all('/a', {some handler function})`

File: client/src/components/QAhelp/QArequests.js [hunk 3]
```diff
@@ -1,88 +1,32 @@
...
-    data: {
-      routing: ['qa', 'questions', 'helpful'],
-      param: [qID.toString(), { question_id: qID.toString() }]
-    }
-  };
 
-  axios.put(`/`, config)
-    .then((res) => res)
-    .catch((err) => console.log(err, 'unsucessful put req'));
+module.exports.postA = (qID, aObj) => {
```
[the remark this answers is not in the document]
Comment (Dalphus) [reply client/src/components/QAhelp/QArequests.js]: got it, changes made

**Pull request merged**